openwisp_controller/config/apps.py

```diff
@@ -172,6 +175,21 @@ def enable_cache_invalidation(self):
             DeviceChecksumView.invalidate_checksum_cache,
             dispatch_uid='invalidate_checksum_cache',
         )
+        post_save.connect(
+            devicegroup_change_handler,
+            sender=self.devicegroup_model,
+            dispatch_uid='invalidate_devicegroup_cache_on_devicegroup_change',
+        )
+        pre_save.connect(
+            devicegroup_change_handler,
+            sender=self.org_model,
+            dispatch_uid='invalidate_devicegroup_cache_on_organization_change',
+        )
+        pre_save.connect(
```

Comment:
why pre save here and above? shouldn't it be post save?

Reply:
Initially I used pre_save just in case organization slug or common name changes. But it won't matter because we are invalidating using celery, so we won't be able to catch those scenarios without additional code.